java/core/src/java/org/apache/orc/impl/ReaderImpl.java

```diff
@@ -365,6 +364,14 @@ public ReaderImpl(Path path, OrcFile.ReaderOptions options) throws IOException {
     this.schema = OrcUtils.convertTypeFromProtobuf(this.types, 0);
   }
 
+  protected FileSystem getFileSystem() throws IOException {
```

Comment:
Two calls to getFileSystem() will do this 2x - have this init this.fileSystem inline.

Reply:
Also use the options.filesystem if it provided

Reply:
fixed it.